ui-admin/src/util/tableUtils.tsx

```diff
@@ -1,23 +1,101 @@
 import React, { HTMLProps, useState } from 'react'
-import { flexRender, Header, Table } from '@tanstack/react-table'
+import { Column, flexRender, Header, Table } from '@tanstack/react-table'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faCaretDown, faCaretUp, faColumns } from '@fortawesome/free-solid-svg-icons'
 
+/**
+ * Returns a debounced input react component
+ * Adapted from https://tanstack.com/table/v8/docs/examples/react/filters
+ */
+function DebouncedInput({
+  value: initialValue,
+  onChange,
+  debounce = 200,
+  ...props
+}: {
+  value: string
+  onChange: (value: string) => void
+  debounce?: number
+} & Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'>) {
```

Comment:
Another way to reference props for builtin components is through `JSX.InstrinsicElements`
```suggestion
} & Omit<JSX.IntrinsicElements['input'], 'onChange'>) {
```

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d952cf91cf1bf44f412df927cf48a18670c0ca5d/types/react/index.d.ts#L3253